docs/topic-guides/tls.txt

```diff
@@ -0,0 +1,68 @@
+.. _tls:
+
+##########################
+ Transport Layer Security
+##########################
+
+**Transport layer security** (TLS) is a protocol for applying encryption
+to network communications. It prevents the data being transmitted from
+being modified or read while it is in transit. Determined can be set up
+to use TLS for all inbound connections to the master. That means that
+all CLI and WebUI connections will be secured by TLS, as well as
+connections from agents and tasks to the master. Communications within
+the cluster that occur as part of :ref:`multi-machine training
+<multi-gpu-training>` are not protected by TLS, nor are proxied
+connections from the master to a TensorBoard or notebook.
+
+**************************
+ Determined Configuration
+**************************
+
+Master
+======
+
+In order to :ref:`configure <master-configuration>` the master to use
+TLS, set the ``security.tls.cert`` and the ``security.tls.key`` options
+to paths to a TLS certificate file and a key file.
+
+When TLS is in use, the default TCP port for the master to listen on if
+none is specified is 8443 rather than 8080.
+
+.. note::
+
+   If the master's certificate is not signed by a widely recognized root
+   certificate, then the configured certificate file must contain a full
+   certificate chain that goes all the way to a root certificate.
+
+Agents
```

Comment:
Should we add a section for "Tasks" and explicitly note that no configuration is necessary / TLS will be injected automatically? etc.